hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/hevm/HederaEvmTransactionResult.java

```diff
@@ -127,10 +141,37 @@ public static HederaEvmTransactionResult resourceExhaustionFrom(
                 null,
                 null,
                 Bytes.wrap(reason.name()),
-                Collections.emptyList());
+                Collections.emptyList(),
+                null);
     }
 
     public boolean isSuccess() {
         return abortReason == null && revertReason == null && haltReason == null;
     }
+
+    private static @Nullable ContractStateChanges allStateAccessesFrom(@NonNull final MessageFrame frame) {
+        return stateChangesFrom(frame, true);
+    }
+
+    private static @Nullable ContractStateChanges stateReadsFrom(@NonNull final MessageFrame frame) {
+        return stateChangesFrom(frame, false);
+    }
+
+    private static @Nullable ContractStateChanges stateChangesFrom(
+            @NonNull final MessageFrame frame, final boolean includeWrites) {
+        requireNonNull(frame);
+        final var accessTracker = accessTrackerFor(frame);
+        if (accessTracker == null) {
+            return null;
+        } else {
+            final List<StorageAccesses> accesses;
+            if (includeWrites) {
+                final var worldUpdater = proxyUpdaterFor(frame);
+                accesses = accessTracker.getReadsMergedWith(worldUpdater.pendingStorageUpdates());
+            } else {
+                accesses = accessTracker.getJustReads();
```

Comment:
Are these few lines worth moving to `StorageAccessTracker`?

Reply:
What method signature did you have in mind? I thought a bit and my ideas seemed awkward.

Reply:
Dunno.  Maybe something like
```
enum Updates { READS_ONLY, READS_AND_WRITES };
getStateChanges(Updates,WorldUpdater)
```

????